src/components/OptionsSelector.js

```diff
@@ -83,35 +125,91 @@ const propTypes = {
 const defaultProps = {
     shouldDelayFocus: false,
     onSelectRow: () => {},
+    textInputLabel: '',
     placeholderText: '',
     selectedOptions: [],
     headerMessage: '',
     canSelectMultipleOptions: false,
     hideSectionHeaders: false,
-    disableArrowKeysActions: false,
     hideAdditionalOptionStates: false,
     forceTextUnreadStyle: false,
     showTitleTooltip: false,
     shouldFocusOnSelectRow: false,
     autoFocus: true,
+    shouldShowConfirmButton: false,
+    confirmButtonText: undefined,
+    maxParticipantsReached: false,
+    maxParticipantsReachedMessage: undefined,
+    onConfirmSelection: () => {},
+    shouldTextInputAppearBelowOptions: false,
+    shouldShowOfflineMessage: false,
+    footerContent: undefined,
+    optionHoveredStyle: styles.hoveredComponentBG,
     shouldShowOptions: true,
+    disableArrowKeysActions: false,
+    isDisabled: false,
 };
 
 class OptionsSelector extends Component {
     constructor(props) {
         super(props);
 
-        this.handleKeyPress = this.handleKeyPress.bind(this);
+        this.updateFocusedIndex = this.updateFocusedIndex.bind(this);
+        this.scrollToIndex = this.scrollToIndex.bind(this);
         this.selectRow = this.selectRow.bind(this);
-        this.viewableItems = [];
         this.relatedTarget = null;
 
+        const allOptions = this.flattenSections();
         this.state = {
-            focusedIndex: 0,
+            allOptions,
+            focusedIndex: this.props.shouldTextInputAppearBelowOptions ? allOptions.length : 0,
```

Comment:
Why does `shouldTextInputAppearBelowOptions` have an impact on the focused index?

Reply:
Because if the text input is below the options, then that's where the user logically starts their arrow navigation. Pressing up brings them to the bottom of the list and pressing down brings them to the top of the list.

Reply:
Not sure if I understand that in practice. Would we want an inverted list in this case? What happens when the list is really long, overflowing, and there's a text input below. Would pressing "up" scroll the user to the bottom of a non inverted list because a text input is displayed under it?

Reply:
p.s. not gonna block on this - just a thought I had.